README.md

```diff
@@ -4,111 +4,214 @@ An API for validating survey schemas.
+- "https://raw.githubusercontent.com/"
+- "onsdigital.uk"
+- "localhost"
+
+Also when using a URL from GitHub you can only validate schemas from the ONSdigital organisation and only against repos with the owner "ONSdigital".
+
+### Running against eQ Runner
+
+Also once you have validator running it can be used to run against eQ runner (`https://github.com/ONSdigital/eq-questionnaire-runner`).
+If you have runner spun up you can from within the root of runner run:
+``` shell
+make validate-test-schemas
 ```
+This script will run validator on the test runner schemas.
 
+### Running the Ajv (server) version of validator
 
-## Testing
+Then run the Ajv server:
+``` shell
+make start-ajv
+```
 
-By default, all schemas in the `tests/schemas/valid` directory will be evaluated as part of the unit tests.
-Any errors in these schemas will cause a failure.
+This defaults to running on port `5002` set `AJV_VALIDATOR_PORT` in your .env file if you need to change this.
 
-To run the app's unit tests:
+Running the Ajv server returns either an empty json response when the questionnaire is valid, or a response containing an "errors" key. The errors are ordered by their path length and with first error message being the deepest path into the schema and should represent the best match for the questionnaire which has been posted.
 
+To stop the Ajv server:
+``` shell
+make stop-ajv
 ```
+
+### Testing and running against local schemas
+
+By default, all schemas in the `tests/schemas/valid` and `tests/schemas/invalid` directories will be evaluated as part of the unit tests. Any errors in these schemas will cause a failure.
+
+To run the app's unit tests:
+``` shell
 make test-unit
 ```
 
-To run the app's unit tests and ajv validator tests:
+Make sure you don't already have Ajv running on localhost:5002 by running `lsof -i tcp:5002` if you do make a note of the PID and then run `kill -9 PID`, replacing "PID" with the process id from the previous command.
 
+Run the Ajv validator tests:
+``` shell
+make test-ajv
 ```
+
+To run the app's unit tests and Ajv validator tests:
+``` shell
 make test
 ```
 
-To test the apps functionality:
-```
+#### Test the local validator app against runner schemas
+
+Spin validator up with:
+``` shell
 make run
 ```
 
-Then, in another terminal window/tab, navigate to a checked out copy of eq-survey-runner:
+Then, in another terminal, navigate to a checked out copy of `https://github.com/ONSdigital/eq-questionnaire-runner` and run:
+``` shell
+make validate-test-schemas
 ```
-make test
+This will run the validator against all runner test schemas.
+
+Or you can run it against a specific runner schema, to do this:
+- set the following vars passing them into the following command:
+    - `SCHEMA_PATH` to the path of the schema file (defaults to `./schemas/test/en/`)
+    - `SCHEMA` to the schema file name without the `.json`
+- then run:
+``` shell
+make validate-test-schema SCHEMA=test_checkbox SCHEMA_PATH=./schemas/
```

Comment:
A few things here:
-  You can pass only the SCHEMA name var and the validator script will default to the `./schemas/test/en/` in Runner. So: `make validate-test-schema SCHEMA=<schema_name>` (that's the most practical way of using it)
- In the example the correct path to make it work in Runner is `SCHEMA_PATH=./schemas/test/en/`
- It is for any schema (you don't say it is an example) so should be something like:  `make validate-test-schema SCHEMA=<schema_name> SCHEMA_PATH=./schemas/test/en/` unless you want to give an example but that should be called out.

Reply:
Maybe it wasn't clear but I was trying to give an example of how you can use the command and what params can be passed in. Thats why I mentioned what the path param defaulted to above it.
I changed the path so that I could demonstrate passing something other than `./schemas/test/en/` into it as that is the default. Ive added added some extra wording to make this more clear.